src/containers/BooksForm.js

```diff
@@ -1,44 +1,98 @@
 import React from 'react';
 import { v4 as uuidv4 } from 'uuid';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { addBook } from '../actions/index';
 
-const BooksForm = () => {
-  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
-  const renderCategories = categories.map((item) => (
-    <option key={uuidv4()}>{item}</option>
-  ));
-
-  return (
-    <div>
-      <button type="button" className="addBook btn btn-primary" data-toogle="modal" data-target="#exampleModal">
-        Add a book!
-      </button>
-      <div className="modal fade" id="exampleModal" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
-        <div className="modal-dialog" role="document">
-          <div className="modal-content">
-            <div className="modal-header">
-              <h5 className="modal-title" id="exampleModalLabel">Add to Bookstore</h5>
-              <button type="button" className="close" data-dismiss="modal" aria-label="Close">
-                <span aria-hidden="true">&times;</span>
-              </button>
-            </div>
-            <div className="modal-body">
-              <div className="dropdown">
-                <select className="custom-select" id="inputGroupSelect01">
-                  <option selected>Choose Category</option>
-                  {renderCategories}
-                </select>
-                <div className="dropdown-menu" aria-labelledby="dropdownMenuButton" />
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+
+    this.myRef = React.createRef();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleDrop = this.handleDrop.bind(this);
+  }
+
+  handleDrop() {
+    this.setState({ title: this.inputNode.value, category: this.myRef.current.value });
+  }
+
+  handleChange(e) {
+    e.preventDefault();
+    /* eslint-disable */
+    this.props.addBook(this.state);
+  }
```

Comment:
- [ ] Perhaps naming this as handleSubmit would be better. In addition, after adding the book consider unsetting state for both title and category.